examples/hsm-example.cpp

```diff
@@ -162,68 +34,32 @@ int main(void)
     // Don't hardcode the pin in your application, this is just for demonstration purposes
     std::string pin("1234");
     HsmEngine hsmEngine(id, modulePath, tokenLabel, pin);
-    std::vector<uint8_t> message = utility::fromHex("deadbeef");
 
-    /************** ECC key generation and ECDSA **************/
+    /************** ECC key generation **************/
     std::vector<uint8_t> keyIDECC{};
     std::string keyLabelECC("ecc-key-label");
-    auto ecdsaDigestType = DigestTypes::SHA512;
     ECCSpec ecspec;
-    auto eccPrivKey =
-            AsymmetricPrivateKey::generateKeyOnHSM(hsmEngine, ecspec, keyLabelECC, keyIDECC);
-    auto ecdsaSigFormat = ECDSASignatureFormat::ASN1_SEQUENCE_OF_INTS;
+    // Keypair containing private and public key
+    auto eccPrivKey = AsymmetricKeypair::generateKeyOnHSM(hsmEngine, ecspec, keyLabelECC, keyIDECC);
 
-    /* The argument hashFunction is optional. Default is SHA256
-     * The default signature format is ECDSASignatureFormat::ASN1_SEQUENCE_OF_INTS */
-    auto ECDSAsignature = ecdsaSign(eccPrivKey, ecdsaDigestType, ecdsaSigFormat, message);
-
-    /* Use-Case 1: You want to check your own signature.
-     * We can use here the private key, as it also contains the public key.
-     * In MoCOCrW the AsymmetricPrivateKey is a specialisation of AsymmetricPublicKey. Thus
-     * we do an implicit upcast here.
-     */
-    ecdsaVerify(eccPrivKey, ecdsaDigestType, ecdsaSigFormat, ECDSAsignature, message);
-
-    /* Use-Case 2: The public key used for verification is stored in the HSM */
+    // Just the public key can be loaded explicitly
```

Comment:
Remove `Just` because then you mention later on that you can load a handle of the private key.